### PR TITLE
add update unit tests

### PR DESCRIPTION
**Pull request by myousfi96**

### CODE REVIEW

File: prSHARK/backends/github.py [hunk 1]
```diff
@@ -336,14 +336,14 @@ def parse_pr_list(self, prs):
             new_pr.is_locked = pr['locked']
             new_pr.lock_reason = pr['active_lock_reason']
             new_pr.is_draft = pr['draft']
-            new_pr.created_at = dateutil.parser.parse(pr['created_at'])
-            new_pr.updated_at = dateutil.parser.parse(pr['updated_at'])
+            new_pr.created_at = dateutil.parser.parse(pr['created_at']).replace(tzinfo=None)
```
Comment (atrautsch) [inline prSHARK/backends/github.py]: Could you shortly explain why the change is needed? Is the timezone info missing from the API?

File: tests/test_github.py [hunk 2]
```diff
@@ -206,3 +206,74 @@ def test_pr_list(self, mock_request):
...
+
+        self.assertEqual(len(pr.pull_request_system_ids), 2)
+        self.assertEqual(len(prr), 1)
+        self.assertEqual(len(prrc), 1)
+        self.assertEqual(len(prc), 1)
+        self.assertEqual(len(pre), 1)
+        self.assertEqual(len(prccf), 1)
+
+    @patch('prSHARK.backends.github.Github._send_request', side_effect=mock_return)
+    def test_update_with_changes(self, mock_request):
```
Comment (atrautsch) [inline tests/test_github.py:250]: Could you write a short comment what you expect to happen, what you test here?
Is it just one additional pull request or was a comment added?

**Review by atrautsch — CHANGES_REQUESTED** [review]

File: prSHARK/backends/github.py (hunk 1 above)
Comment (myousfi96) [reply prSHARK/backends/github.py]: when we save the PR the timezone information will not be saved in MongoDB. So when we compare new_pr and mongo_pr it will show a difference. For example : {'new_value': datetime.datetime(2011, 1, 26, 19, 1, 12, tzinfo=tzutc()), 'old_value': datetime.datetime(2011, 1, 26, 19, 1, 12)}

Comment (atrautsch) [reply prSHARK/backends/github.py]: Alright, in that case we should just check whether the API returns UTC so that we can remove the timezone and save it to the DB. The important thing is that we have UTC in the DB.

Comment (myousfi96) [reply prSHARK/backends/github.py]: Yes, we will have UTC time in the DB.  The statement replace(tzinfo=None) will only remove the info.

Comment (atrautsch) [reply prSHARK/backends/github.py]: You can probably add a small datetime helper function here. My suggestion is analogous to the actionSHARK.

```suggestion
            new_pr.created_at = dateutil.parser.parse(pr['created_at']).astimezone(pytz.UTC).replace(tzinfo=None)
```

File: prSHARK/backends/github.py [hunk 3]
```diff
@@ -615,9 +615,13 @@ def check_diff(self, old, new, ex_path):
...
 
         """
+        if self.pr_id not in self.pr_diff:
+            self.pr_diff[self.pr_id] = False
+
         if old:
-            diff = DeepDiff(t1=old.__dict__, t2=new.__dict__, exclude_paths=ex_path)
+            diff = DeepDiff(t1=old.to_mongo().to_dict(), t2=new.to_mongo().to_dict(), exclude_paths=['_id', ex_path])
             if diff:
+                print(diff)
```
Comment (atrautsch) [inline prSHARK/backends/github.py]: debug?

**Review by atrautsch — CHANGES_REQUESTED** [review]
Looks good! Only thing left is the timezone.
